scan/scan.go

```diff
@@ -237,11 +247,51 @@ func (s *sourcescanner) walkSource(conn redis.Conn, path string, f os.FileInfo,
 		return nil, nil
 	}
 
+	var dfData DummyFile
+	dummyFile := GetConfig().DummyFiles
+
 	d := new(filedata)
 	d.path = path[len(GetConfig().Repository):]
-	d.size = f.Size()
-	d.modTime = f.ModTime()
 
+	if dummyFile {
+		fi, _ := os.Stat(d.path)
+		if fi.Size() > 1048576 {
```

Comment:
Magic numbers appearing in the middle of the code are usually bad. One way to improve this specific piece of code is to add to the `util` package (or even `util` subpackage) something like:
```go
type ByteSize int64

const (
	_           = iota
	KB ByteSize = 1 << (10 * iota)
	MB
	GB
	TB
	PB
)
```
Therefore your check will end up looking like this instead: `fi.Size() > 1 * utils.MB`.

That being said, do we really want to enforce / hardcode a limit? What if someday we'll have much bigger (and valid) files to parse? One way I see is to use a [json.Decoder](https://golang.org/pkg/encoding/json/#NewDecoder) instead. It takes an `io.Reader` and therefore will only read the necessary portion of files (in most cases, way less than 1MB). It will also have the advantage of being faster.